RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -5956,17 +5958,38 @@ int meatReserve()
 		}
 	}
 	
+	//how much do we reserve for a zeppelin ticket?
+	if(my_level() >= 11 && internalQuestStatus("questL11Ron") < 5 && item_amount($item[Red Zeppelin Ticket]) < 1)
+	{	//the copperhead part tries for a priceless diamond, but if it's over without getting one
+		if( (get_property("questL11Shen") == "finished" || $location[the copperhead club].turns_spent >= 15) && item_amount($item[priceless diamond])< 1)
+			reserve_zeppelin += 5000 * npcStoreDiscountMulti();
+	}	
+	
+	//how much do we reserve for palindome photographs?
+	if(my_level() >= 11 && internalQuestStatus("questL11Palindome") < 1)
+	{	
+		if(item_amount($item[Photograph Of A Red Nugget]) < 1)
+			reserve_palindome += 500;
+		if(item_amount($item[Photograph Of God]) < 1)
+			reserve_palindome += 500;
+	}
+	
 	//how much do we reserve for unlocking mysterious island?
 	if(get_property("lastIslandUnlock").to_int() < my_ascensions())		//need to unlock island
 	{
-		reserve_island += 1500;		//3 vacations. no need to count script. we don't pull it or get it prematurely.
+		//do count scrips. they may have been pulled manually, and one optional property does pull them
+		//if we have more than a single one or reasons to assume it's not for a UV compass
+		if(item_amount($item[Shore Inc. Ship Trip Scrip]) > 0 && (item_amount($item[Shore Inc. Ship Trip Scrip]) >= 2 || internalQuestStatus("questL11Desert") >= 1 || possessEquipment($item[UV-resistant compass]) || !auto_can_equip($item[UV-resistant compass]) || possessEquipment($item[Ornate Dowsing Rod]) || isActuallyEd()))
```

Comment:
the logic strikes me as a bit convoluted. I think it should be something like (pseudocode)
```
int price_vacation = 500;
if(path == surprising fist)
{
price_vacation = 5;  //yes really. just 5 meat each
}
int scrip_needed = 3;
if(various checks for compass need with comments on why we are doing them)
{
scrip_needed += 1;
}
int scrip_missing = scrip_needed - min(scrip_needed,item_amount($item[Shore Inc. Ship Trip Scrip]);

reserve_island += scrip_missing * price_vacation;
```

Reply:
code with scrips for later

		int price_vacation = 500;
		if(my_path() == "Way of the Surprising Fist")
		{
		price_vacation = 5;  //yes really. just 5 meat each
		}

		//do count scrips. they may have been pulled manually, and one optional property does pull them
		int scrip_available = item_amount($item[Shore Inc. Ship Trip Scrip]);
		
		if(scrip_available > 0 && 					//first verify if one will get used for the UV compass
		internalQuestStatus("questL11Desert") < 1 && !isActuallyEd() &&	//need to do desert
		!possessEquipment($item[Ornate Dowsing Rod]) &&			//with rod, L11_getUVCompass() doesn't get compass
		!possessEquipment($item[UV-resistant compass]) && 
		auto_can_equip($item[UV-resistant compass]))
		{
			scrip_available -= 1;	//one of the scrips could get used for compass first
		}
		reserve_island += price_vacation * (3 - max(3,scrip_available));	//3 scrips needed